extensions/shared/number-control.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import {
+	NumberControl as BlockEditorNumberControl,
+	__experimentalNumberControl as ExperimentalNumberControl,
+	TextControl,
+} from '@wordpress/components';
+
+/**
+ * This uses the experimental NumberControl from the block editor where available,
+ * otherwise it falls back to a standard TextControl, limited to numbers.
+ *
+ * @param {any} props - the NumberControl component props
+ * @returns {object} - NumberControl component
+ */
+export const NumberControl =
```

Comment:
I think we should export the component as the default. The file name already specifies the component name.

```suggestion
const NumberControl =
```